src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,70 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { setCurrentTodo } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const [currentUser, setCurrentUser] = useState<null | User>(null);
+
+  useEffect(() => {
+    if (currentTodo) {
+      setIsLoading(true);
+      getUser(currentTodo?.userId)
+        .then(setCurrentUser)
+        .catch(() => {
+          throw new Error('error loading user');
```

Comment:
Throwing an error inside a catch block without handling it will cause the application to crash. Consider logging the error or displaying a user-friendly message instead.